Add troubleshooting steps to free-tier connect doc

ctauchen opened the pull request:
Clarified symptoms and solutions for common problems.
Added note regarding Goldmane enablement to OSS upgrade
flow.

<!--- PR title format: [GH#<gh-issue-id>][DOCS#<jira-issue-id>]: <short-description-of-the-pr> --->

Product Version(s):
<!--- Specify which versions of Calico, Calico Enterprise, and Calico Cloud your PR applies to. -->
OSS 3.30+
CC all
Issue:
<!--- Add a link to the Jira ticket or GitHub issue, if applicable. --->

Link to docs preview:
<!--- Add direct link(s) to the exact page(s) with updated content from the preview build. --->
Free tier troubleshooting section: https://deploy-preview-2380--tigera.netlify.app/calico-cloud/free/connect-cluster-free#troubleshooting
OSS upgrade section: https://deploy-preview-2380--tigera.netlify.app/calico/latest/operations/upgrading/kubernetes-upgrade#upgrading-an-installation-that-uses-the-operator


SME review:
- [ ] An SME has approved this change. 
<!--- SME approval is required to merge a PR except for changes that do not impact the meaning of the docs. --->

DOCS review:
- [ ] A member of the docs team has approved this change.
<!-- The Docs team must review and approve all changes. -->

Additional information:
<!--- Optional: Include additional context or expand the description here.--->

Merge checklist:
<!--- Mandatory for docs team members before merging code --->
- [ ] Deploy preview inspected wherever changes were made 
- [ ] Build completed successfully
- [ ] Test have passed 


<!--- After you open your PR, ask for review from docs team:
  For community authors: tag @tigera/docs to ask for a review when your PR is ready --->

tmjd commented on calico-cloud/free/connect-cluster-free.mdx:
```diff
@@ -122,25 +122,60 @@ Try the [Calico Cloud Free Tier quickstart guide](quickstart.mdx) to create a cl
+  A few minutes after you apply the `Goldmane` resource, you will begin to see data in your dashboards and Service Graph.
 
-You can find the address of the Calico Cloud management cluster in the `ManagementClusterAddr` field of the `ManagementClusterConnection` resource:
+### Connection status remains **Disconnected** after applying the connection manifest
```
The connection manifest should have failed to apply if they were still on 3.29. You could add a Troubleshooting symptom of "Applying the connection manifest reported errors".

ctauchen replied:
I'm not quite sure what the problem is. Here, I describe a general problem (cluster is still showing disconnected) and two things that may be causing the problem. 

In the 3.29 case, yes, the connection manifest ultimately failed to apply. Which is what the output on lines 149-152 shows.

Perhaps we're just cutting the problem in different ways.

tmjd replied:
I was suggesting adding a new troubleshooting problem of "Applying the connection manifest reported errors" because the user should recognize that as a problem before they even check that their status remains Disconnected.

So someone looking at the troubleshooting might not recognize that their problem is identified under "Connection status remains **Disconnected** after applying the connection manifest" because they realized that it errored when applying the connection manifest.

This would result in having the same solution under 2 different sections because the user realized the problem at 2 different places in the user journey.

tmjd commented on calico_versioned_docs/version-3.30/operations/upgrading/kubernetes-upgrade.mdx:
```diff
@@ -44,19 +44,13 @@ owned resources being garbage collected by Kubernetes.
 
 ## Upgrading an installation that was installed using Helm
 
-1. Update the Helm chart to the latest version:
```
Did you mean to include this in this PR? I would think this is an important step and shouldn't be removed also.

ctauchen replied:
Yeah, that was not intended. Fixing.

ctauchen committed:
Add troubleshooting steps to free-tier connect doc

Clarified symptoms and solutions for common problems.
Added note regarding Goldmane enablement to OSS upgrade
flow.
